op-chain-ops/cmd/celo-migrate/ancients.go

```diff
@@ -152,6 +155,12 @@ func transformBlocks(ctx context.Context, in <-chan RLPBlockRange, out chan<- RL
 			for i := range blockRange.hashes {
 				blockNumber := blockRange.start + uint64(i)
 
+				if blockNumber != 0 && blockNumber != prevBlockNumber+1 {
+					return fmt.Errorf("gap found between ancient blocks numbered %d and %d. Please delete the target directory and repeat the migration with an uncorrupted source directory.", prevBlockNumber, blockNumber)
+				}
```

Comment:
I don't think this is going to work, If you are doing a second pre-migrate and therefore `blockNumber` is not 0 then this condition will be triggered.

Reply:
Good catch!

Reply:
If `startBlock` can be 0, which I think it can for the first block then `prevBlockNumber` will overflow.